src/main/java/com/github/dockerjava/core/LocalDirectorySSLConfig.java

```diff
@@ -53,13 +56,20 @@ public SSLContext getSSLContext() {
                 String certpem = new String(Files.readAllBytes(Paths.get(certPemPath)));
                 String capem = new String(Files.readAllBytes(Paths.get(caPemPath)));
 
-                SslConfigurator sslConfig = SslConfigurator.newInstance(true);
-                sslConfig.securityProtocol("TLSv1.2");
-                sslConfig.keyStore(CertificateUtils.createKeyStore(keypem, certpem));
-                sslConfig.keyStorePassword("docker");
-                sslConfig.trustStore(CertificateUtils.createTrustStore(capem));
+                String kmfAlgorithm = AccessController.doPrivileged(getSystemProperty("ssl.keyManagerFactory.algorithm",
+                    KeyManagerFactory.getDefaultAlgorithm()));
+                KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(kmfAlgorithm);
+                keyManagerFactory.init(CertificateUtils.createKeyStore(keypem, certpem), "docker".toCharArray());
 
-                return sslConfig.createSSLContext();
+                String tmfAlgorithm = AccessController.doPrivileged(getSystemProperty("ssl.trustManagerFactory.algorithm",
+                    TrustManagerFactory.getDefaultAlgorithm()));
```

Comment:
why it needed?

Reply:
@KostyaSha any thoughts on my last comment?